access/types/forms.py

```diff
@@ -322,12 +322,18 @@ def grade(self):
                 prev = i
                 i, ok, p = self.grade_field(i, field)
                 points += p
-                if not ok:
+                if self.exercise["feedback"]:
+                    self.fields[self.field_name(prev, field)].grade_points = field.get('points', 0)
+                elif not ok:
                     error_fields.append(self.field_name(prev, field))
                     gname = self.group_name(g)
                     if gname not in error_groups:
                         error_groups.append(gname)
             g += 1
+
+        if self.exercise["feedback"]:
+            points = self.exercise["max_points"]
```

Comment:
Change `self.exercise["max_points"]` into `self.exercise.get("max_points", 0)` just in case max_points are not set.